event_reporter/utils/app_revision.go

```diff
@@ -46,6 +63,64 @@ func GetOperationRevision(a *appv1.Application) string {
 	return revision
 }
 
+func GetOperationSyncRevisions(a *appv1.Application) []string {
+	var revisions []string
+	if a != nil {
+		// this value will be used in case if application hasn't resources, like empty gitsource
+		revisions = getRevisions(RevisionsData{
+			Revision:  a.Status.Sync.Revision,
+			Revisions: a.Status.Sync.Revisions,
+		})
+
+		if a.Status.OperationState != nil && a.Status.OperationState.Operation.Sync != nil {
+			revisions = getRevisions(RevisionsData{
+				Revision:  a.Status.OperationState.Operation.Sync.Revision,
+				Revisions: a.Status.OperationState.Operation.Sync.Revisions,
+			})
+		} else if a.Operation != nil && a.Operation.Sync != nil {
+			revisions = getRevisions(RevisionsData{
+				Revision:  a.Operation.Sync.Revision,
+				Revisions: a.Operation.Sync.Revisions,
+			})
+		}
+	}
+
+	return revisions
+}
+
+// for monorepo support: list with revisions where actual changes to source directory were committed
+func GetOperationChangeRevisions(a *appv1.Application) []string {
+	var revisions []string
+	if a != nil {
```

Comment:
could you please check if app is nil then return empty array, it will not have so much embedded conditions